AUV/auv_viewer_processing/auv_processing.py

```diff
@@ -0,0 +1,725 @@
+
+    return lower_lft_latlon, lower_rgt_latlon, upper_lft_latlon, upper_rgt_latlon, center_latlon
+
+def distance_btw_latlon(lft_latlon, rgt_latlon):
+    """
+    returns the distance in meters between 2 lat lon coordinates
+    """
+    return vincenty(lft_latlon, rgt_latlon).meters
+
+def geotiff_list_metadata(geotiff_list):
+    """
+    generate a list of gdal metadata for all geotiff found for one dive
+    """
+    header = "campaign_code,dive_code,image_filename,longitude,latitude,"\
+            "image_width,depth_sensor,altitude_sensor,depth,"\
+            "sea_water_temperature,sea_water_salinity,"\
+            "chlorophyll_concentration_in_sea_water,backscattering_ratio,"\
+            "colored_dissolved_organic_matter,time,cluster_tag,up_left_lon,"\
+            "up_left_lat,up_right_lon,up_right_lat,low_right_lon,low_right_lat,"\
+            "low_left_lon,low_left_lat".split(",")
+
+    geotiff_list_metadata = []
+    for row, geotiff in enumerate(geotiff_list):
+        geotiff_list_metadata.append( {k: '' for k in header}) # initialise dict for every row
+        geotiff_coordinate = geotiff_corner_coordinates(geotiff)
+        geotiff_width      = distance_btw_latlon(geotiff_coordinate[0][0:2],
+                                            geotiff_coordinate[1][0:2])
+
+        geotiff_list_metadata[row]["image_filename"] = os.path.splitext(os.path.basename(geotiff))[0]
+        geotiff_list_metadata[row]["up_left_lon"]    = geotiff_coordinate[2][0]
+        geotiff_list_metadata[row]["up_left_lat"]    = geotiff_coordinate[2][1]
+        geotiff_list_metadata[row]["up_right_lon"]   = geotiff_coordinate[3][0]
+        geotiff_list_metadata[row]["up_right_lat"]   = geotiff_coordinate[3][1]
+        geotiff_list_metadata[row]["low_left_lon"]   = geotiff_coordinate[0][0]
+        geotiff_list_metadata[row]["low_left_lat"]   = geotiff_coordinate[0][1]
+        geotiff_list_metadata[row]["low_right_lon"]  = geotiff_coordinate[1][0]
+        geotiff_list_metadata[row]["low_right_lat"]  = geotiff_coordinate[1][1]
+        geotiff_list_metadata[row]["image_width"]    = geotiff_width
+
+    return geotiff_list_metadata
+
+def _csv_track_dive_path(dive_path):
+    """
+    retrieves the csv track file path for one dive
+    """
+    for file in os.listdir(dive_path):
+        if file.endswith("track_files"):
+            track_folder_path = os.path.join(dive_path, file)
+            break
+
+    for file in os.listdir(track_folder_path):
+        if file.endswith("_latlong.csv"):
+             return os.path.join(track_folder_path, file)
+
+    # if file not found
+    logger.error('CSV Track file not found - Process of campaign aborted')
+    raise Exception('CSV Track file not found')
+
+def read_track_csv(dive_path):
+    """
+    import the data found in the csv track file for one dive. This version handles
+    the cluster tag info. Some improvements could be done to generate automatically
+    the converters variable
+    """
```

Comment:
This could be done in a line or two using `pandas`. See http://pandas.pydata.org/pandas-docs/stable/generated/pandas.read_csv.html#pandas.read_csv